### ISSUE TITLE
add Windows 10 1607 (Anniversary Update) support

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#1987]
This is another major update.

Quick tests show that x86 and x64 do work already, thanks to #1854.  However, wow64 crashes in release for as-yet undetermined reasons.  In debug it asserts but this should be innocuous and we should remove this assert:

```
<Application C:\derek\hello.exe (5092).  Internal Error: DynamoRIO debug check failure: d:\drmemory_package\dynamorio\core\win32\ntdll.c:621 *(tgt + 0x18) == 0xea

ntdll!NtTestAlert:
77c1f6c0 b8ac010200      mov     eax,201ACh
77c1f6c5 ba3023c377      mov     edx,offset ntdll!Wow64SystemServiceCall (77c32330)
77c1f6ca ffd2            call    edx
77c1f6cc c3              ret

0:000> U ntdll!Wow64SystemServiceCall 
ntdll!Wow64SystemServiceCall:
77c32330 ff251812cc77    jmp     dword ptr [ntdll!Wow64Transition (77cc1218)]
0:000> U poi(77cc1218)
58787000 ea097078583300  jmp     0033:58787009
```

So it's not the wrapper itself but the transition code.  We should just relax that as we ignore whatever is at the callee anyway.  But why does it crash then in release?

**Comment by derekbruening** [comment]
Xref https://github.com/DynamoRIO/drmemory/issues/1906

**Comment by derekbruening** [comment]
New system calls in ntoskrnl:

```
# awk '{print $NF}' 10.0.10586.20-x86/syscalls | sort > /tmp/wold
# awk '{print $NF}' 10.0.14393.0-x86/syscalls | sort > /tmp/wnew
# diff /tmp/wold /tmp/wnew
124a125,126
> NtCommitRegistryTransaction
> NtCommitRegistryTransaction
194a197,198
> NtCreateRegistryTransaction
> NtCreateRegistryTransaction
444a449,450
> NtOpenRegistryTransaction
> NtOpenRegistryTransaction
572a579,580
> NtQuerySecurityPolicy
> NtQuerySecurityPolicy
678a687,688
> NtRollbackRegistryTransaction
> NtRollbackRegistryTransaction
698a709,710
> NtSetCachedSigningLevel2
> NtSetCachedSigningLevel2
```

wow64 and x64 diffs are identical.

**Comment by derekbruening** [comment]
The wow64 crash is due to instr_is_wow64_syscall() failing to identify the transition code.  I added this to #1854 as it is an unfortunate case of having to rely too much on particular, fragile code sequences.

**Comment by derekbruening** [comment]
Running debug DR with the bbsize client now handles notepad's Save As dialog as a quick sanity test after these:

9b7e44e #1987 Win10-1607: update winsysnums to find split usercalls
651abf5 #1987 Win10-1607: add artificial version and syscall numbers
8e6b428 #1987 Win10-1607: support new wow64 transition code
99530ca #1988: fix nop padding bug with MPX prefixes

I relaxed the wow64 syscall test, though at the risk that it will think a regular call* is a syscall: we'll see a curiosity in that case.  This is calculated as a lower risk than the risk that a hardcoded test just breaks again in the next win10 update.

Additional testing will come with Dr. Memory.  Closing this.
